2_adventures/001/kotp/04_automated_testing_example/lib/fizz_buzz.rb

```diff
@@ -0,0 +1,19 @@
+module FizzBuzz
+  def self.play(number)
+    result = ''
+    result << 'Fizz' if number.divisible_by?(3)
+    result << 'Buzz' if number.divisible_by?(5)
+    result.empty? ? number : result
+  end
+end
+
+module FB_Tools
```

Comment:
Thanks @lenn4rd .  I agree.

It was educational rather than practical.  And demonstrated as part of a journey in refactoring.

Here is a further refactored version of FizzBuzz derived through conversations... and still not the final version.

```
class FizzBuzz
  def initialize(rule = { 3 => 'Fizz', 5 => 'Buzz' }, rules: rule)
    @rules = rules
  end

  # to_s takes an optional argument of a range otherwise simply plays the game
  # from 1 to 100
  def to_s(range = (1..100))
    range.map { |n| what_is(n) }.join("\n")
  end

  def what_is(integer)
    result = @rules.each_key.map do |k|
      (integer % k).zero? ?  @rules[k] : ''
    end.join
    result.empty? && integer || result
  end
end
```

As we can see, it is indeed "noise" and contrived.